sdk/python/kfp/dsl/component_factory.py

```diff
@@ -69,22 +70,34 @@ def _python_function_name_to_component_name(name):
     return name_with_spaces[0].upper() + name_with_spaces[1:]
 
 
-def make_index_url_options(pip_index_urls: Optional[List[str]]) -> str:
+def make_index_url_options(pip_index_urls: Optional[List[str]],
+                           pip_trusted_hosts: Optional[List[str]]) -> str:
     """Generates index url options for pip install command based on provided
     pip_index_urls.
 
     Args:
         pip_index_urls: Optional list of pip index urls
+        pip_trusted_hosts: Optional list of pip trust hosts
 
     Returns:
         - Empty string if pip_index_urls is empty/None.
-        - '--index-url url --trusted-host url ' if pip_index_urls contains 1
+        - '--index-url url ' if pip_index_urls contains 1
         url
-        - the above followed by '--extra-index-url url --trusted-host url '
+        - the above followed by '--extra-index-url url '
         for
         each next url in pip_index_urls if pip_index_urls contains more than 1
         url
 
+        - pip_trusted_hosts was added later. In this case, if pip_trusted_hosts is None or empty
+        - the above followed by '--trusted-host url '
+        for
+        each url in pip_index_urls
+
+        - if pip_trusted_hosts is greater than 0
+        - the above followed by '--trusted-host url '
+        for
+        each url in pip_trusted_hosts
```

Comment:
```suggestion
        - otherwise if one or more pip_trusted_hosts is provided, then this takes precedence over pip_index_urls and a pip flag  '--trusted-host url'  is appended for each url in pip_trusted_hosts.
```